### PR TITLE
Final commit seur 1.3

### PR DESCRIPTION
**Pull request by jcriadop**
I hope now you can merge commits. Kind regards.

### CODE REVIEW

**Comment by gaillafr** [comment]
The module does not work: https://www.dropbox.com/s/c9a8zuffjrbw1dz/Screenshot%202015-06-24%2013.47.05.png?dl=0

**Comment by jcriadop** [comment]
Hi François, I check module and install me ok, I can't reply error. I send you by email original module, so you can check it.

You can check line 46 on our fork:

https://github.com/jcriadop/seur/blob/master/seur.php

Kind regards.

**Comment by Quetzacoalt91** [comment]
Hi @jcriadop,

This error has also been reported here: http://forge.prestashop.com/browse/PNM-3517
Please have a look on my comment in the following link to have more information.

Best regards

**Comment by jcriadop** [comment]
Hi @Quetzacoalt91 

Added new commit to pull request to solve visibility (changed from private to public).

Best regards.

**Comment by Quetzacoalt91** [comment]
Yep, it is better like this. :)
I just saw that changing the visibility was not enough, because the parent function needs one parameter.

File: ajax/createLabelAjax.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+include('../../../config/config.inc.php');
+include('../../../init.php');
+
+if (!class_exists('SeurLib')) include(_PS_MODULE_DIR_ . 'seur/classes/SeurLib.php');
+if (!class_exists('Pickup')) include(_PS_MODULE_DIR_ . 'seur/classes/Pickup.php');
+
+$order_data = (array) $_GET;
+
+if (Tools::getValue('token') != Tools::getAdminToken('AdminOrders' . (int) (Tab::getIdFromClassName('AdminOrders')) . (int) (Tools::getValue('id_employee'))))
+    Tools::display404Error();
```
Comment (Quetzacoalt91) [inline ajax/createLabelAjax.php]: Please note this function is deprecated from PS 1.5.5.0. You should use `Controller::getController('PageNotFoundController')->run();` as written in the comment.

**Comment by jcriadop** [comment]
Hi, deprecated function changed.

Kind regards.

File: views/templates/admin/AdminSeur.tpl [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+							</td>
+							
+						{/foreach}
+						{/if}
+						</tr>
+					
+					{foreachelse}
+					<div class="bootstrap">
+						<div class="module_error alert alert-danger">
+							<button data-dismiss="alert" class="close" type="button">�</button>
```
Comment (Quetzacoalt91) [inline views/templates/admin/AdminSeur.tpl]: This file is not encoded in UTF-8. A character in this line is not properly displayed.

**Comment by Quetzacoalt91** [comment]
Hi,

When I go to the order details (made with SEUR), I get the following fatal error:
capture du 2015-07-07 16 08 43

**Comment by jcriadop** [comment]
Hi. We can't reply this error. We check it on localhost and many environments, and module works fine at all. I think problem is on your server. Can you send us your phpinfo information by email?

Kind regards.

**Comment by CamSi** [comment]
Hi @jcriadop ,
@Quetzacoalt91  sent you an email with the php information!
Kind regards

**Comment by jcriadop** [comment]
Hi @CamSi, phpinfo received and checked. I sent some alternative by email, but it seems to be a protocol bug (not module code bug), so our options are very limited.
Kind regards.

**Comment by Quetzacoalt91** [comment]
You still have an option: A better management of the exception which could be thrown.

I have just tested something: Catch every possible SoapFaults in classes/Rate.php, classes/Town.php etc. by adding this piece of code:

``` php
catch (SoapFault $e)
{
    //$e->displayMessage();
    return false;
}
```

By doing this, I can at least reach my order page. But of course, there are another errors to deal with in a second time.

A fatal error is the worst case for the merchants and must be avoided when possible.

Best regards

**Comment by jcriadop** [comment]
Hi @Quetzacoalt91, added "return false" to "classes" directory files to avoid fatal error, pending better solution.

Kind regards.

**Comment by Quetzacoalt91** [comment]
Hi @jcriadop,

Unfortunately the issue still occurs. As you can see in [the screenshot I sent you](https://github.com/PrestaShop/seur/pull/5#issuecomment-119215314), we are dealing with an uncaught exception: SoapFault.
In consequence I did not modified the existing catch, but I added another one for Soap. The rsult would be:

``` php
catch (PrestaShopException $e)
{
    //$e->displayMessage();
    return false;
}
catch (SoapFault $e)
{
    //$e->displayMessage();
    return false;
}
```

**Comment by jcriadop** [comment]
Hi @Quetzacoalt91 , added second catch.
Kind regards.

**Comment by Quetzacoalt91** [comment]
Hi @jcriadop,

This commit removes the fatal error, that's better. But there are still many warnings to fix:
capture du 2015-07-20 12 05 17

And I don't know if it is related, but the design seems to be broken on the order details:
capture du 2015-07-20 12 05 54

Kind regards

**Pull request merged**